services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py

```diff
@@ -263,6 +263,21 @@ async def mark_project_tasks_as_aborted(self, project: ProjectAtDB) -> None:
                 .values(state=StateType.ABORTED)
             )
 
+    @log_decorator(logger=logger)
+    async def mark_project_tasks_as_pending(
+        self, project_id: ProjectID, tasks: List[NodeID]
+    ) -> None:
+        # block all pending tasks, so the sidecars stop taking them
+        async with self.db_engine.acquire() as conn:
+            await conn.execute(
+                sa.update(comp_tasks)
+                .where(
+                    (comp_tasks.c.project_id == str(project_id))
+                    & (comp_tasks.c.node_id.in_([str(t) for t in tasks]))
+                )
+                .values(state=StateType.PENDING)
+            )
```

Comment:
not sure why you ask that?
- only other service that modifies the value is the sidecar, and only if the value is set to PENDING.

Reply:
What I was thinking is what would it happen if the `director-v2` and the `sidecar` both touch the table at the same time. I was thinking if this race condition could create any problems.

Reply:
If there might be an issue the following might help.

To avoid race conditions, I would expose an endpoint on `director-v2` to update the state of the task form the `sidecar`. And then I would properly handle locking on this table with some mechanism provided by SQL or distributed locking.

Reply:
yes, so the current situation of the sidecar modifying these entries is not new, and I aim to fix this in a separate PR.
ideally only the director-v2 should modify the comp_tasks table. Now, as you know it is not feasible until the dynamic-sidecar is out. Even then we will still have the old services to somehow handle.